### PR TITLE
 Improved tests for QueryResponseSender

### PR DESCRIPTION
**Pull request by iainmcgin**
- Don't rely on mocks
- Test based on actual broadcasts sent

### CODE REVIEW

File: bbq/javatests/java/com/google/bbq/QueryResponseSenderTest.java [hunk 1]
```diff
@@ -24,56 +26,87 @@
...
-    Context mockContext;
-    @Mock
-    Intent mockIntent;
 
-    BroadcastQuery mockQuery;
-
-    private QueryResponseSender underTest;
+    private BroadcastQuery mQuery;
+    private QueryResponseSender mResponseSender;
+    private Intent mIntent;
```
Comment (nerdyverde) [inline bbq/javatests/java/com/google/bbq/QueryResponseSenderTest.java]: Looks like `mIntent` is never used

File: bbq/javatests/java/com/google/bbq/QueryResponseSenderTest.java [hunk 2]
```diff
@@ -24,56 +26,87 @@
...
 import com.google.bbq.Protobufs.BroadcastQuery;
 import com.google.bbq.Protobufs.BroadcastQueryResponse;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
```
Comment (nerdyverde) [inline bbq/javatests/java/com/google/bbq/QueryResponseSenderTest.java]: There seem to be a few unused import statements with these changes

Comment (iainmcgin) [reply bbq/javatests/java/com/google/bbq/QueryResponseSenderTest.java]: I really need to create a checkstyle config for the tests. The main reason why it doesn't currently look at test code is that I don't want to force some things, like Javadoc on all public methods, on test code where it's not useful.

File: bbq/javatests/java/com/google/bbq/QueryResponseSenderTest.java (hunk 1 above)
Comment (iainmcgin) [reply bbq/javatests/java/com/google/bbq/QueryResponseSenderTest.java]: Removed, thanks!

**Pull request merged**